app/design/frontend/base/default/template/boltpay/prefetch_shipping_quote.phtml

```diff
@@ -24,48 +24,53 @@
  * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
 
-if($this->isBoltActive()) {
+if($this->isBoltActive() && $this->isPrefetchable()) {
```

Comment:
I thought that as well at first, but if you look closer, it actually calls the PHP script before it makes the AJAX request. My code does the same thing, but it uses the data to determine whether or not to make the prefetch request whereas the old code always makes a request to get the GeoIP details and it calls prefetch. See here for my notes https://screencast.com/t/r8wx4a1Xtug

Reply:
The logic is flawed here at two levels.  We must first keep in mind that the goal is to simply gracefully handle errors from freegeoip.  That is handled by checking what is returned in a call to it.  In this code, `$locationInfo['code'] == 200` or any expected key in this array can be checked to confirm a successful call.

The problems are as follows:

1.)  Success or failure to calling freegeoIP should not determine whether Magento prefetch is called.  *Magento prefetch should always be called.*  Prefetch should be using other data points like customer default shipping or billing address when available.  It should only fall back on GeoIP when all other means have been exhausted.  The code you present here will bypass all prefetching on GeoIP failure, and this is not correct.

2.) Lack of  `currentZipCode` returned from freegeoIP should not be used to determine whether or not to call prefetch for the same reason as above and because zipcode can be country specific where some zip codes may not exist for certain countries and addresses in the database.  Even when it is the case that no zipcode data is available, the country and the region are almost always sufficient to create the shipping and tax estimate.

Zip code should only be used in any client-side DOM/Javascript code for filling form data correctly before triggering events.